### PR TITLE
LCD_Unit_test

### PR DESCRIPTION
**Pull request by ahmedsalama09**
LCD_Task tested.. Application found at msg_app.c  file
to: Mohamed Tarek

### CODE REVIEW

**Comment by BassamSharkawy-Sprints** [comment]
What do you mean  by adding files by upload - Rename

File: Slama/LCD.c [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	
+	/*check if LCD initialized before*/
+	if(gu8_LcdInitState == LCD_NOT_INITIALIZED)
+	{
+		if(gu8_init1st == ZERO)
+		{
+			/*set pins as output
+		PB0 : ENABLE PIN , PB1: RS PIN , PB2 : RW PIN*/
+		st_DIO_Cfg_t st_dio_init = {.GPIO = LCD_CTRL_PORT , .pins = (LOWER_NIBBLE) , .dir = OUTPUT};
+		DIO_init(&st_dio_init);
```
Comment (YehiaSprints) [inline Slama/LCD.c]: You need to check the error returned from DIO_init

File: Slama/LCD.c [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+		
+		gu8_init1st = 1;
+		}
+	
+	/*init_lcd for 4 bits mode*/
+	if(gu8_commandFlag == ZERO)
+	{
+	
+	LCD_sendCommand(FOUR_BITS_DATA_MODE_ACTIVATED);
```
Comment (YehiaSprints) [inline Slama/LCD.c]: You need to check the error status here also.
Don't leave a function that returns an error without checking its error status.

File: Slama/LCD_UnitTest.c [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+*@output: Error Value
+*/
+void Test_LCD_DATA(void)
+{
+    /*Test 1 Description : Happy scenario everything done as expected
+    *Expected Output : E_OK
+    *Output : E_OK
+    *Test State: Passed
+    */
+    printf("%d",LCD_DATA(5));
```
Comment (YehiaSprints) [inline Slama/LCD_UnitTest.c]: What if the data cannot be displayed on the LCD, I think (5) won't be displayed on the LCD and the test should return invalid data.

File: Slama/LCD_UnitTest.c [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+*@output: Error Value
+*/
+void Test_LCD_sendCommand(void)
+{
+    /*Test 1 Description : Happy scenario everything done as expected
+    *Expected Output : E_OK
+    *Output : E_OK
+    *Test State: Passed
+    */
+    printf("%d",LCD_sendCommand(5));
```
Comment (YehiaSprints) [inline Slama/LCD_UnitTest.c]: What if the command (5) is wrong?

File: Slama/LCD_UnitTest.c [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+*/
+void Test_LCD_displayChar(void)
+{
+    /*Test 1 Description : Happy scenario everything done as expected
+    *Expected Output : E_OK
+    *Output : E_OK
+    *Test State: Passed
+    */
+    LCD_init();
+   printf("%d",LCD_displayChar(5));
```
Comment (YehiaSprints) [inline Slama/LCD_UnitTest.c]: The same here the data isn't valid.

File: Slama/LCD_UnitTest.c (hunk 3 above)
Comment (YehiaSprints) [reply Slama/LCD_UnitTest.c]: I think you didn't implement the LCD_DATA function.

**Comment by YehiaSprints** [comment]
No Comments from Mohammed Tarek.

**Comment by BassamSharkawy-Sprints** [comment]
Please apply Yahia's/M.Tarek's reviews and merge.

**Comment by BassamSharkawy-Sprints** [comment]
Please apply Yahia's/M.Tarek's reviews and merge.

**Pull request merged**